### ISSUE TITLE
ENOENT: no such file or directory, realpath

### ISSUE THREAD

**Issue by muuvmuuv** [just-jeb/angular-builders#1730]
## Describe the Bug

Angular with latest TypeScript semver range supports direct import.

```
An unhandled exception occurred: ENOENT: no such file or directory, realpath '/Users/marvin/Developer/M8FINDER/M8FINDER/.angular/vite-root/app'
See "/private/var/folders/nb/bwd6_n3d3y583d365z6mcb200000gn/T/ng-Q5hK1L/angular-errors.log" for further details.
```

> https://github.com/just-jeb/angular-builders/blob/master/packages/common/src/load-module.ts#L84-L87

Must be changed to:

```ts
return import(modulePath).then((module) => module.default);
```

## Minimal Reproduction

**Note:** Our policy is that issues reported without a reproduction will be closed immediately and then reopened once a reproduction has been provided. Please respect the developers of this project by doing this. We give of our personal time to work on this project and would rather be spending our time fixing or enhancing the library than chasing down badly described or unreproducible issues.
Please delete this note once you have read it.

## Environment

<pre><code>
Libs
- @angular/core version: 17.3.2
- @angular-devkit/build-angular version: 17.3.2
- @angular-builders/custom-esbuild version: 17.1.1

For Tooling issues:
- Bun version: 1.0.35
- Platform:  Mac

Others:

</code></pre>

**Comment by just-jeb** [comment]
Do you know what's the minimal requirement for Angular/TS here @muuvmuuv ?

**Comment by muuvmuuv** [comment]
I already tried to but its hard to say. We would need tests for that to check against lower semver ranges. But since it no longer works on my end between these and the previous version I would say this is the range.

**Comment by muuvmuuv** [comment]
Would it make sense to grab that here after v18 release since TS version is now high enough?

**Comment by just-jeb** [comment]
Absolutely, I'd say if anything it must be part of v18 release (since as part of breaking changes we can require minimal TS version). 
Do you think you'd have time to make a PR with this change into [v18 PR](https://github.com/just-jeb/angular-builders/pulls)?

**Comment by muuvmuuv** [comment]
Yeah ofc. But I don't think I can write tests for that, Jest is kinda picky with mixed import syntax so it wont be ez.. maybe a custom test suite in shell-script with tsc?

**Comment by muuvmuuv** [comment]
Just something I discovered developing in angular-builders is that "@lerna-lite/run" is missing in order to do something like: `yarn lerna run --scope @angular-builders/common test` OR how do you execute scripts inside a package?

**Comment by just-jeb** [comment]
> maybe a custom test suite in shell-script with tsc?
Yeah, e2e test are probably the way to go. We already have custom shell scripts for e2e tests, for example [this](https://github.com/just-jeb/angular-builders/blob/master/packages/custom-webpack/scripts/ci.sh) one for custom-webpack builder or [this](https://github.com/just-jeb/angular-builders/blob/master/packages/jest/scripts/ci.sh) one for the Jest builder. 

I think that the existing e2e tests for builders will implicitly cover the change (coz they depend on `@angular-builders/common` and use `loadModule`) but feel free to add more and/or add a proprietary ci shell script for `common`.

> how do you execute scripts inside a package

We use `yarn workspaces foreach` for multiple packages or `yarn workspace` for a single package (like [here](https://github.com/just-jeb/angular-builders/blob/master/package.json#L30)). In most cases this should be enough.

**Comment by muuvmuuv** [comment]
Ok, this requires way more work, it does work in my szenario because my repo has type=module and other stuff set to newer settings, but it does not in here because a lot still requires cjs syntaxes. I guess every example and package must first be updated to be esm only or something like that but idk if that even works with Angular. I tried setting stuff to type=module but that comes with another error "Unknown file extension ".ts" and so on ^^

**Comment by just-jeb** [comment]
Just to make it clear: is it a bug or an enhancement? We have examples with both `csj` and `esm` which are working for `custom-webpack` and `custom-esbuild` builders.

**Comment by just-jeb** [comment]
The earlier conversation suggested this may have resolved with a TS update. If you're still hitting it on current versions, a minimal repro would help.
